bin/sozo/src/commands/execute.rs

```diff
@@ -56,6 +45,69 @@ pub struct ExecuteArgs {
     pub transaction: TransactionOptions,
 }
 
+#[derive(Debug)]
+pub struct CallArgs {
+    pub tag_or_address: ResourceDescriptor, // Contract address or tag
+    pub entrypoint: String,                 // Entrypoint to call
+    pub calldata: Option<String>,           // Calldata to pass to the entrypoint
+}
+
+impl std::str::FromStr for CallArgs {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let s = s.trim();
+
+        let parts: Vec<&str> = s.split(',').collect();
+        if parts.len() < 2 {
+            return Err(anyhow!("Invalid call format. Expected format: <CONTRACT_NAME>,<ENTRYPOINT_NAME>,<ARG1>,<ARG2>,..."));
+        }
+
+        let entrypoint = parts[1].trim();
+        if entrypoint.is_empty() {
```

Comment:
please write a unit test for this function.

Reply:
Hey @kariy, can you tell me the steps to write a unit test for this function? I am kinda new to this.

Reply:
Hey @glihm , I am actually kinda getting an error running the test 
This errors are like this:
` Compiling zstd v0.13.2
   Compiling nextest-runner v0.65.0
    Finished `release` profile [optimized] target(s) in 2m 27s
  Installing /home/manush2005/.cargo/bin/cargo-nextest
   Installed package `cargo-nextest v0.9.82` (executable `cargo-nextest`)
manush2005@Manush:/mnt/c/Users/Manush/OneDrive/Desktop/Manush/dojo$ cargo nextest run --all-features -p sozo
    Updating crates.io index
    Updating git repository `https://github.com/dojoengine/dojo`
error: failed to get `dojo-lang` as a dependency of package `scarb v2.8.4 (https://github.com/dojoengine/scarb?rev=7eac49b3e61236ce466e712225d9c989f9db1ef3#7eac49b3)`
    ... which satisfies git dependency `scarb` (locked to 2.8.4) of package `dojo-test-utils v1.0.0 (/mnt/c/Users/Manush/OneDrive/Desktop/Manush/dojo/crates/dojo/test-utils)`    
    ... which satisfies path dependency `dojo-test-utils` (locked to 1.0.0) of package `dojo-utils v1.0.0 (/mnt/c/Users/Manush/OneDrive/Desktop/Manush/dojo/crates/dojo/utils)`   
    ... which satisfies path dependency `dojo-utils` (locked to 1.0.0) of package `katana v1.0.0 (/mnt/c/Users/Manush/OneDrive/Desktop/Manush/dojo/bin/katana)`

Caused by:
  failed to load source for dependency `dojo-lang`

Caused by:
  Unable to update https://github.com/dojoengine/dojo?rev=6725a8f20af56213fa7382aa1e158817f3ee623c#6725a8f2

Caused by:
  failed to fetch into: /home/manush2005/.cargo/git/db/dojo-10cac2e09298cf35

Caused by:
  network failure seems to have happened
  if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
  https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli

Caused by:
  SSL error: unknown error; class=Ssl (16)`

Reply:
Also , @glihm  and @kariy , can you please highlight some changes I need to make in the current code. It is diffcult for me to grasp as it is my frist time contributing to dojo

Reply:
@Manush-2005 how are you trying to compile the code locally?

You should follow the `CONTRIBUTING.md` guide to help you out.
If you have any question, you're welcome in the dojo discord and happy to have a call to show you the flow if you still have complications to build.